lib/default.nix

```diff
@@ -573,4 +573,21 @@ in {
     inherit pkgs;
     inherit (pkgs.buildPackages.buildPackages) runCommand;
   };
+
+  # Here we try to figure out which qemu to use based on the host platform.
+  # This guess can be overridden by passing qemuSuffix
+  qemuByHostPlatform = hostPlatform:
+    # I'd prefer this was a dictionary lookup, with a fall through into abort,
+    # that would make this more readable I guess.  I think there is some similar
+    # mapping somewhere in haskell.nix
+    if hostPlatform.isAarch32
+    then "arm"
+    else if hostPlatform.isAarch64
+    then "aarch64"
+    else abort "Don't know which QEMU to use for hostPlatform ${hostPlatform.config}. Please provide qemuSuffix";
+
+  # How to run ldd when checking for static linking
+  lddForTests = if haskellLib.isCrossHost && stdenv.hostPlatform.isLinux && (stdenv.hostPlatform.isAarch32 || stdenv.hostPlatform.isAarch64)
+    then "${pkgs.pkgsBuildBuild.qemu}/bin/qemu-${haskellLib.qemuByHostPlatform stdenv.hostPlatform} ${pkgs.iconv}/bin/ldd"
+    else "ldd";
```

Comment:
This is nearly the same code as we have in the `linux-cross` file, no? I think we need some better comment that explains why we need qemu here, and why we need to run `ldd` through qemu; why do we need `ldd` (from `iconv`?) to check for static linking? There should be a better way? Can't `patchelf` tell us?

Reply:
> This is nearly the same code as we have in the linux-cross file, no?

Yes, it is was in [overlays/linux-cross.nix](https://github.com/input-output-hk/haskell.nix/pull/1911/files#diff-aaee47856780386b3c12f71bf72918285c094751b57623fb07cce1765ba08a99L1).  I moved it here to make it accessible to the tests.

> from iconv?

I got the idea to use `iconv` by from running `which aarch66-unknown-gnul-linux-ldd` in a dev shell.  It pointed to the `iconv` derivation in `/nix/store`.  I was a bit puzzled as to why it comes from `iconv`.  I'll see if there is a more sensible way to depend on it.

> to check for static linking? There should be a better way? Can't patchelf tell us?

Yes.  All the tests currently use `ldd` and look at the error output. I'll give `patchelf` a go.

Reply:
Going with `lddForTests = "${pkgs.pkgsBuildBuild.glibc.bin}/bin/ldd"` for now.  We can look into `readelf` as an alternative in another PR.